### ISSUE TITLE
Have alias function names for dot functions

### ISSUE THREAD

**Issue by TimJones** [rosineygp/mkdkr#66]
I realise that `..` and `...` were chosen because "It creates a beautiful code style like yaml", but beauty is very subjective and dot functions are not exactly intuitive (outside of [Brainfuck](https://en.wikipedia.org/wiki/Brainfuck)). Especially considering having `..` enter *into* a container when in a shell it *leaves/goes up* a directory.
I would suggest having aliases e.g. `..` could also be `dkr_exec` and `...` could be `dkr_start`. These are just examples/suggestions and should probably be discussed a little more.

**Comment by rosineygp** [comment]
If I change will be something like Dockerfile syntax

```Dockerfile
my-job:
    $(mkdkr)
    FROM alpine
    RUN ps -ef
```
But It will broken a lot of things and I can apply only in major change.
If it really bother you, try this path ;)
```diff
diff --git a/.mkdkr b/.mkdkr
index bde5156..79eecaf 100755
--- a/.mkdkr
+++ b/.mkdkr
@@ -81,7 +81,7 @@ SHELL = /bin/bash
 MKDKR_BRANCH_NAME=$MKDKR_BRANCH_NAME
 MKDKR_BRANCH_NAME_SLUG=$MKDKR_BRANCH_NAME_SLUG
 
-define . =
+define mkdkr =
   . .mkdkr
   \$(eval MKDKR_JOB_NAME=\$(shell . .mkdkr; .... \$(@)))
   \$(eval MKDKR_JOB_STDOUT="${TMP_DIR}/\${MKDKR_JOB_NAME}.log")
@@ -104,7 +104,7 @@ EOF
 }
 
 # shellcheck disable=SC1036
-...() {
+FROM() {
   OLDIFS=$IFS
   IFS='⠀' # Braille Pattern Blank (U+2800)
   TYPE="${1}"
@@ -119,7 +119,7 @@ EOF
 
   _remove_running_job
 
-  pretty "cyan" "... ${TYPE} ${IMAGE} ${ARGS[*]}"
+  pretty "cyan" "FROM ${TYPE} ${IMAGE} ${ARGS[*]}"
 
   if [ "${TYPE}" == "service" ]; then
     docker run --rm -d \
@@ -152,10 +152,10 @@ EOF
 }
 
 # shellcheck disable=SC1036
-..() {
+RUN() {
   CONTAINER_NAME="job_${MKDKR_JOB_NAME}"
 
-  pretty "green" ".. ${*}"
+  pretty "green" "RUN ${*}"
 
   docker exec -i ${CONTAINER_NAME} ${MKDKR_SHELL:-sh} -c "$*" | tee "${TMP_DIR}/${MKDKR_JOB_NAME}.log"
   EXIT_CODE="${PIPESTATUS[0]}"
```

Not tested and probably broken :-1: 

I already try alias in function, but not worked.

**Comment by ochi** [comment]
```
$ cat Makefile.aliases 
include $(shell bash .mkdkr init)

simple:
    @$(.)
    ... alpine
    .. echo "hello mkdkr!"

alias:
    @$(.)
    FROM alpine
    RUN echo "hello mkdir!"
```
```
$ make -f Makefile.aliases alias

include

update, alias: commitlint, repos: https://github.com/rosineygp/mkdkr_commitlint.git, checkout: master, file: main.mk
remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
From ssh://github.com/rosineygp/mkdkr_commitlint
 * branch            master     -> FETCH_HEAD
Already up-to-date.
Current branch master is up to date.
HEAD is now at 10f5d25 chore: fixed mkdkr_docker version v0.7.0

alias

... job alpine 
c0cdec32c3c42aa2ff1ac0fd4f515dd07993f8ae8754dab5470ce4b662642bce
.. echo hello mkdir!
hello mkdir!
.
c0cdec32c3c4
```
achieved by simply extending what's already in place...
```
$ git diff
diff --git a/.mkdkr b/.mkdkr
index 1e63bcd..04f6f60 100755
--- a/.mkdkr
+++ b/.mkdkr
@@ -249,3 +249,8 @@ if [ "$1" == "init" ]; then
 
   echo "$(_header) $(_remote_include)"
 fi
+
+FROM(){ ... $@; }
+RUN(){ .. $@; }
+
```
I agree with @TimJones - 2 dots are "slightly" confusing - feel free to implement however you feel is a proper way :)

**Comment by rosineygp** [comment]
I almost finish this "first" version, all stuffs that I needed is implemented.
So I will start to break compatibility or aliases.

```Makefile
old_simple:
  @$(.)
  ... alpine
  ... service nginx
  ... privileged docker:19
  .. my command
```

```Makefile
simple:
  @$(.)
  JOB alpine
  SERVICE nginx
  PRIVILEGED docker:19
  RUN my command
  PULL <future: pull files>
  PUSH <future: push files>
  IGNORE_ERRORS <future: do not stop if it failed>
```

I guess is a better common syntax with more possibilities.

I will not apply alias now, because I have to rewrite or double all tests, I had a lot of problems with passing parameters in bash.

So I prefer one function, without alias.

**Comment by rosineygp** [comment]
So... I wrote some sketches and I have the following result and some drops.

Uppercase no.

Just inspired in Dockerfile... all steps isn't very good in this project.

Take a look in following previews:
https://github.com/rosineygp/mkdkr/tree/preview/1
https://github.com/rosineygp/mkdkr/tree/preview/2

TL;DR

Preview 1
```Makefile
simple:
	@$(dkr)
	instance job alpine
	run echo "hello mkdkr!"

service:
	@$(dkr)
	service nginx
	instance alpine --lservice_$$MKDKR_JOB_NAME:nginx
	run apk add curl
	run curl -s nginx

dind:
	@$(dkr)
	dind docker:19
	run docker build -t rosiney/pylint .
```

Preview 2 (just add : at end of function)

```Makefile
simple:
	@$(dkr)
	instance: job alpine
	run: echo "hello mkdkr!"

service:
	@$(dkr)
	service: nginx
	instance: alpine --lservice_$$MKDKR_JOB_NAME:nginx
	run: apk add curl
	run: curl -s nginx

dind:
	@$(dkr)
	dind: docker:19
	run: docker build -t rosiney/pylint .
```

@thebouv what do you think about?

**Comment by TimJones** [comment]
@rosineygp big 👍 from me for preview2

**Comment by rosineygp** [comment]
@TimJones everything done!!
